src/mo_interpreter/interpret.ml

```diff
@@ -119,6 +120,27 @@ struct
       Printf.eprintf "%s: execution error, %s\n" (Source.string_of_region at) msg
   let rec run () =
     if not (Queue.is_empty q) then (yield (); run ())
+
+  let tmp : (unit -> unit) Queue.t = Queue.create ()
+  let bounce work =
+    (* add work to *front* of queue *)
+    Queue.transfer q tmp;
+    Queue.add work q;
+    Queue.transfer tmp q
+
+  let interval = 128
+  let count = ref interval
+  let trampoline3 f x y z =
+    if !Flags.ocaml_js then
```

Comment:
Curious, what's the down side of trampolining the interpreter in native OS?

Reply:
Dunno, but I image its more expensive than a tailcall.